core/runtime.rs

```diff
@@ -316,38 +327,68 @@ impl Drop for JsRuntime {
   }
 }
 
-impl JsRuntime {
+impl <const FOR_SNAPSHOT: bool> JsRuntimeImpl<FOR_SNAPSHOT> {
   const STATE_DATA_OFFSET: u32 = 0;
   const MODULE_MAP_DATA_OFFSET: u32 = 1;
 
-  /// Only constructor, configuration is done through `options`.
-  pub fn new(mut options: RuntimeOptions) -> Self {
+  pub fn new_for_snapshot(mut options: RuntimeOptions, runtime_snapshot_options: RuntimeSnapshotOptions) -> JsRuntimeImpl<true> {
```

Comment:
Could this be like
```rust
struct JsRuntimeImpl<const FOR_SNAPSHOT: bool = false> {}

impl JsRuntimeImpl<true> {
  fn new() -> Self {
    Self {}
  }
}

impl JsRuntimeImpl<false> {
  fn new() -> Self {
    Self {}
  }
}
```

Reply:
I think so.. with the aliases this would be pretty clean as well (`JsRuntime::new` and `JsRuntimeForSnapshot::new`).

Reply:
Done